Home assitant setup (#25)
ClemensElflein/openmower.de: content/en/docs/contributions/Home_assitant_MQTT/_index.md

---
title: Add OpenMower in Home Assitant with MQTT
linkTitle: Home Assistant
description: >
  Add OpenMower in Home Assitant with MQTT.
---

Author: Etienne@discord , DocGalaxyBlock@discord , JuergenLeber@discord

### Prerequisites

> - **Home assistant**

Reviewer: ```suggestion
- **Home Assistant**
```

- **[Mosquitto broker addons](https://github.com/home-assistant/addons/tree/master/mosquitto)**

### Step 0: Configure mqtt in Home Assistant
Go to to addons settings and add in "Logins"
 ```bash
 - username: openmoweruser
   password: openmowerpass
```
Of course choose your own password.

> ### Step 2: Setup MQTT bridqe on openmower

Reviewer: Ideally keep the same capitalization, here and below
+ typo

```suggestion
### Step 2: Setup MQTT bridge on OpenMower
```

Enable MQTT on mower_config.txt
```bash
sudo nano /boot/openmower/mower_config.txt
```

> You need change IP adress,user and password.

Reviewer: ```suggestion
You need to change IP address, user and password.
```

> Exemple:

Reviewer: ```suggestion
Example:
```

```

Test if service is started:
```bash
sudo systemctl status openmower-mqtt-bridge.service
```
Should return something like this:
```bash
● openmower-mqtt-bridge.service - OpenMower MQTT Bridge
     Loaded: loaded (/etc/systemd/system/openmower-mqtt-bridge.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2023-07-08 10:14:55 BST; 1 day 5h ago
       Docs: man:mosquitto.conf(5)
             man:mosquitto(8)
    Process: 5977 ExecStartPre=/bin/mkdir -m 740 -p /var/log/mosquitto (code=exited, status=0/SUCCESS)
    Process: 5978 ExecStartPre=/bin/chown mosquitto /var/log/mosquitto (code=exited, status=0/SUCCESS)
    Process: 5979 ExecStartPre=/bin/mkdir -m 740 -p /run/mosquitto (code=exited, status=0/SUCCESS)
    Process: 5980 ExecStartPre=/bin/chown mosquitto /run/mosquitto (code=exited, status=0/SUCCESS)
   Main PID: 5981 (mosquitto)
      Tasks: 1 (limit: 4471)
        CPU: 7min 52.689s
     CGroup: /system.slice/openmower-mqtt-bridge.service
             └─5981 /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf

Jul 08 22:43:28 openmower mosquitto[5981]: 1688852608: Connecting bridge (step 1) bridge-to-external-broker (192.168.xxx.xxx:1883)
```

> Now openmower is connect to your HA MQTT addons.

Reviewer: ```suggestion
Now OpenMower is connected to your HA MQTT addons.
```
